src/go/k8s/pkg/resources/statefulset_update_test.go

```diff
@@ -258,3 +258,135 @@ vectorized_cluster_partition_under_replicated_replicas{namespace="kafka",partiti
 	err := ssres.evaluateUnderReplicatedPartitions(context.Background(), &adminURL)
 	require.NoError(t, err)
 }
+
+func Test_sortPodList(t *testing.T) {
+	const clusterName = "sortPodListCluster"
+	cluster := vectorizedv1alpha1.Cluster{ObjectMeta: metav1.ObjectMeta{Name: clusterName}}
+
+	emptyPodList := corev1.PodList{
+		Items: []corev1.Pod{},
+	}
+
+	orderedPodList := corev1.PodList{
+		Items: []corev1.Pod{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-0"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-1"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-2"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-3"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-4"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-5"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-6"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-7"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-8"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-9"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-10"},
+			},
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: clusterName + "-11"},
+			},
```

Comment:
NIT: I wouldn't write this by hand, but rather generate the data and store it in testdata folder. Load them during test and compare results.

Reply:
I like to just have plain data in tests. It's easy to look and see what's happening when something fails.